packages/flutter/lib/src/cupertino/button.dart

```diff
@@ -146,16 +149,19 @@ class _CupertinoButtonState extends State<CupertinoButton> with SingleTickerProv
     _setTween();
   }
 
-  void _handleTapDown(PointerDownEvent event) {
-    _animationController.animateTo(1.0, duration: kFadeOutDuration);
+  Future<Null> _handleTapDown(PointerDownEvent event) async {
+    await _lastAnimation;
+    _lastAnimation = _animationController.animateTo(1.0, duration: kFadeOutDuration);
```

Comment:
How will this mess up? The first tap down will animate from 0.0 to 1.0. The next tap down (and all subsequent once) will animate from 1.0 to 1.0 (which is a no-op after this fix).